demisto_sdk/commands/download/downloader.py

```diff
@@ -95,647 +87,851 @@
+        logger.info("Fetching custom content bundle from server...")
+
         try:
-            verify = (
-                (not self.insecure) if self.insecure else None
-            )  # set to None so demisto_client will use env var DEMISTO_VERIFY_SSL
             self.client = demisto_client.configure(verify_ssl=verify)
-            api_response: tuple = demisto_client.generic_request_func(
-                self.client, "/content/bundle", "GET"
-            )
-            body: bytes = ast.literal_eval(api_response[0])
-            io_bytes = io.BytesIO(body)
-
-            # Demisto's custom content file is of type tar.gz
-            tar = tarfile.open(fileobj=io_bytes, mode="r")
-
-            (
-                content_items_file_names_tuple,
-                scripts_id_to_name,
-            ) = self.find_uuids_in_content_item(tar)
-            self.write_custom_content(
-                content_items_file_names_tuple, scripts_id_to_name
-            )
-            return True
+            api_response: HTTPResponse = demisto_client.generic_request_func(
+                self.client,
+                "/content/bundle",
+                "GET",
+                _preload_content=False,
+            )[0]
 
-        except ApiException as e:
-            self.handle_api_exception(e)
-            return False
-        except MaxRetryError as e:
-            self.handle_max_retry_error(e)
-            return False
         except Exception as e:
-            logger.info(f"Exception raised when fetching custom content:\n{e}")
-            return False
+            if isinstance(e, ApiException) and e.status == 401:
+                logger.error(
+                    f"Server authentication error: {e}\n"
+                    "Please verify that the required environment variables ('DEMISTO_API_KEY', or "
+                    "'DEMISTO_USERNAME' and 'DEMISTO_PASSWORD') are properly configured."
+                )
+
+            elif isinstance(e, MaxRetryError):
+                logger.error(
+                    f"Failed connecting to server: {e}.\n"
+                    "Please verify that the environment variable 'DEMISTO_BASE_URL' is properly configured, "
+                    "and that the server is accessible.\n"
+                    "If the server is using a self-signed certificate, try using the '--insecure' flag."
+                )
+
+            else:
+                logger.error(f"Error while fetching custom content: {e}")
+
+            raise HandledError from e
+
+        logger.info("Custom content bundle fetched successfully.")
+        logger.debug(
+            f"Downloaded content bundle size (bytes): {len(api_response.data)}"
+        )
+
+        loaded_files: dict[str, StringIO] = {}
+
+        with tarfile.open(fileobj=BytesIO(api_response.data), mode="r") as tar:
+            tar_members = tar.getmembers()
+            logger.debug(f"Custom content bundle contains {len(tar_members)} items.")
```

Comment:
Resolving following a Slack conversation, as this can result in very large logs (there can be 1,000+ files, and file names can be very long as they include prefixes like `playbook-` and `automation-`).